docs/customization_extendibility.md

```diff
@@ -0,0 +1,72 @@
+# Extending and Customizing
+For many simpler web applications, `ddev` provides everything you need to successfully provision and develop a web application on your local machine out of the box. More complex and sophisticated web applications, however, often require integration with services beyond the standard requirements of a web and database server. Examples of these additional services are Apache Solr, Redis, Varnish, etc. While `ddev` likely won't ever provide all of these additional services out of the box, it is designed to provide simple ways for the environment to be customized and extended to meet the needs of your application.
+
+## Prerequisite knowledge
+The majority of `ddev`'s customization ability and extensibility comes from leveraging features and functionality provided by [Docker](https://docs.docker.com/) and [Docker Compose](https://docs.docker.com/compose/overview/). Some working knowledge of these tools is required in order to customize or extend the environment `ddev` provides.
+
+## Extending with additional docker-compose files
+Under the hood, `ddev` uses docker-compose to define and run the multiple containers that make up the local environment for a web application. Docker compose supports defining multiple compose files to facilitate [sharing Compose configurations between files and projects](https://docs.docker.com/compose/extends/), and `ddev` is designed to leverage this ability. Additional services can be added to your project by creating additonal docker-compose files in the `.ddev` directory for your project. Any files using the `docker-compose.[servicename].yml` naming convention will be automatically processed by `ddev` to include them in executing docker-compose functionality. A `docker-compose.override.yml` can additonally be created to override any configurations from the main docker-compose file or any service compose files added to your project. 
+
+## Conventions for defining additonal services
+When defining additional services for your project, it is recommended to follow these conventions to ensure your service is handled by `ddev` the same way the default services are.
+- Containers should be follow the naming convention `local-[sitename]-[servicename]`
+- Containers should be provided the following labels:
```

Comment:
We should probably discuss what labels should be required for users defining their own containers.
- I'm pretty sure `site-name` is a requirement.
- I put `approot` and `app-url` in here, but I'm not certain its truly necessary.
- `container-type` is currently required, but I would like to deprecate this label in favor of "com.docker.compose.service", so that we can all define one less label. With #198 this default label provides the same value we were putting in container-type.